### ISSUE TITLE
zavm07 - sp

### ISSUE THREAD

**Issue by Ajaxspeed** [nvbach91/4IZ278-2021-2022-LS#101]
### Driedmeats
E-shop se sortimentem sušeného masa a obdobných pokrmů
### Funkcionality
Po načtení se uživateli zobrazí úvodní stránka s novými a akčními produkty a horní menu s tlačítky pro přihlášení a registraci (v případě přihlášeného uživatel tlačítko přehled účtu a odhlášení). Horní menu dále obsahuje tlačítka s názvy kategorií (hovězí, zvěřinové...), které uživatele přesunu na stránku s produkty příslušné kategorie. Po kliknutí na produkt se zobrazí jeho detail. Z výpisu i deailu bude možné přidat produkt do košíku. Kliknutím na ikonku košíku v horním menu se zobrazí zvolené produkty včetně možnosti nastavení množství. V objednávce dále mohou pokračovat pouze registrovaní uživatelé. Každá objednávka má v sobě uloženy dodací údaje (jeden uživatel může objednávat na více míst) a cenu (pro případ změny cen produktů). Registrovaní uživatelé si dále můžou změnit své údaje, zobrazit historii objednávek a odhlásit se. Některým uživatům můžou být přidělena administrátorská práva. Díky nim je možné pomocí tlačítek v detailu produktu (ostatním uživatelům skryty) vytvářet, upravovat a mazat jednotlivé produkty. Úprava produktů bude ošetřena optimistickým zámkem.

- - -
### Use case

![Use case](https://user-images.githubusercontent.com/6597082/164895609-20ea9929-a8c8-42ed-a4d9-2cd98d4c25b0.png)

- - -
### Databáze
![db_schema](https://user-images.githubusercontent.com/6597082/164895735-63ec3837-6e06-463d-b331-fc4c6b9d5c63.PNG)

*time_edited v tabulce products je ve skutečnosti datový typ datetime (způsobeno nepřesným zobrazení návrhového nástroje)

**Comment by nvbach91** [comment]
nebylo by dobre ukladat ceny produktu i v tabulce ordered_items?

**Comment by Ajaxspeed** [comment]
Souhlasím, model DB byl upraven. Celková cena objednávky se tím pádem vypočte z ceny objednaných položek, není ji nutné uchovávat v rámci objednávky
